File: api/features.go
package api

import "net/http"

type features struct {
	Search     bool `json:"search,omitempty"`
	Extractor  bool `json:"extractor,omitempty"`
	ProxyHTTP  bool `json:"proxyHTTP,omitempty"`
	Popularity bool `json:"popularity,omitempty"`
}

func featuresHandler(features features) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		args{"features": features}.WriteJSON(w)
	}
}

File: rf-ng/src/app/services/features.ts
import { Injectable } from '@angular/core'
import { Router } from '@angular/router'
import { Observable, ConnectableObservable } from "rxjs";
import { APIService } from "./api";
import { TokenService } from "./auth";
import { switchMap, publishReplay, map } from 'rxjs/operators';



interface featuresPayload {
    features: Features
}

export interface Features {
	i18n:       boolean;
	search:     boolean;
	extractor:  boolean;
	proxyHTTP:  boolean;
	popularity: boolean;
}

@Injectable({providedIn: "root"})
export class FeaturesService {
    private features: Observable<Features>;

    constructor(
        private api: APIService,
        private tokenService: TokenService,
    ) {
        var features = this.tokenService.tokenObservable().pipe(
            switchMap(token => this.api.get<featuresPayload>("features").pipe(
                map(p => p.features)
            )),
            publishReplay<Features>(1)
        ) as ConnectableObservable<Features>;
        features.connect();

        this.features = features;
    }

    getFeatures() : Observable<Features> {
        return this.features;
    }
}